[nmstate-0.3] nm vlan/vxlan: Use uuid for VLAN/VxLAN parent

cathay4t opened the pull request:
When parent of VLAN/VxLAN is holding the same name of OVS bridge or OVS
port, NetworkManager will fail with error failed to find interface index
of that parent.

The root cause is NetworkManager try to use user space interface as
VLAN/VxLAN parent.

To workaround that, use profile UUID for `NM.SettingVlan.props.parent`
and `NM.SettingVxlan.props.parent`.

Integration test case included.

ffmancera commented on libnmstate/nm/applier.py:
```diff
@@ -87,6 +85,8 @@ def apply_changes(context, net_state, save_to_disk):
         _create_proxy_ifaces_desired_state(ifaces_desired_state)
     )
 
+    pending_con_profiles = []
```
I am fine with all the changes. I will run the integration tests several time so see if there are random errors. Anyway, could you add a comment here? I assume for the context that `pending_con_profiles` is a list of tuples, the first element of the tuple is the current_profile for that device and the second element is the new_profile. Adding a little comment with this information will be nice. **NOT A MUST**, thank you!

cathay4t replied:
Comment included. Thanks.

cathay4t committed:
nm vlan/vxlan: Use uuid for VLAN/VxLAN parent

When parent of VLAN/VxLAN is holding the same name of OVS bridge or OVS
port, NetworkManager will fail with error failed to find interface index
of that parent.

The root cause is NetworkManager try to use user space interface as
VLAN/VxLAN parent.

To workaround that, use profile UUID for `NM.SettingVlan.props.parent`
and `NM.SettingVxlan.props.parent`.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
